### PR TITLE
fix: request license information from GitLab API

### LINKED ISSUE #6589: fix: GitLab project license property always empty due to missing ?license=true parameter
## Summary

The `gitlab/license` entity property (`RepoPropertyLicense`) is always empty for GitLab repositories because the project fetch in `internal/providers/gitlab/repository_properties.go` does not pass the `?license=true` query parameter required by the GitLab API to include license information in the response.

## Reproduction

Without `?license=true`:
```
GET /api/v4/projects/gitlab-org%2Fgitlab
→ "license": null
```
With `?license=true`:
```
GET /api/v4/projects/gitlab-org%2Fgitlab?license=true
→ "license": {"key": "mit", "name": "MIT License", ...}
```
## Fix

In `getGitLabProject()`, append `?license=true` to the project URL path:

```go
projectURLPath, err := url.JoinPath("projects", url.PathEscape(upstreamID))
// change to:
projectURLPath = projectURLPath + "?license=true"
```

## Impact

- `gitlab/license` property is always `""` for all GitLab repositories
- Any rule relying on `RepoPropertyLicense` or `.Entity.License` for GitLab
  will never get license data
- Discovered during OSPS level-1 rule porting work (#6435)

### To Reproduce

_No response_

### What version are you using?

_No response_

### PR DESCRIPTION
**Pull request by Anni0506**
# Summary

***Provide a brief overview of the changes and the issue being addressed. 
Explain the rationale and any background necessary for understanding the changes. 
List dependencies required by this change, if any.***

Fixes #(related issue)

# Testing

# Summary

This PR fixes an issue where the `gitlab/license` entity property (`RepoPropertyLicense`) is always empty for GitLab repositories. The fetch logic in `internal/providers/gitlab/repository_properties.go` has been updated to correctly pass the `?license=true` query parameter required by the GitLab API to include license details in the response.

Fixes #6589

# Testing

This is a straightforward API query parameter addition. The underlying `gitlab` provider package logic handles the mapping once the license payload is retrieved via the modified URL path.

### CODE REVIEW

**Comment by Anni0506** [comment]
I have read the CLA Document and I hereby sign the CLA

File: internal/providers/gitlab/repository_properties.go [hunk 1]
```diff
@@ -50,6 +50,7 @@ func (c *gitlabClient) getGitLabProject(
 	ctx context.Context, upstreamID string,
 ) (*gitlab.Project, error) {
 	projectURLPath, err := url.JoinPath("projects", url.PathEscape(upstreamID))
+	projectURLPath = projectURLPath + "?license=true"
 	if err != nil {
```
Comment (intelligent-ears) [inline internal/providers/gitlab/repository_properties.go:53]: The `?license=true` is appended before the error check — the code appends the query param before checking if `url.JoinPath` returned an error. If `JoinPath` fails, `projectURLPath` would be an empty string and the append would produce just `?license=true`. The correct order is:

```
projectURLPath, err := url.JoinPath("projects", url.PathEscape(upstreamID))
if err != nil {
    return nil, fmt.Errorf("failed to join URL path for project using upstream ID: %w", err)
}
projectURLPath = projectURLPath + "?license=true"
```

**Review by intelligent-ears — COMMENTED** [review]
This PR doesn't add any tests verifying that the license field actually gets populated.There can be at least a unit test for `gitlabProjectToProperties` that confirms `license` is non-empty when the API response includes a license.

**Comment by evankanderson** [comment]
> This PR doesn't add any tests verifying that the license field actually gets populated.There can be at least a unit test for `gitlabProjectToProperties` that confirms `license` is non-empty when the API response includes a license.

In particular, we should probably update `Test_gitlabClient_FetchAllProperties` in `properties_test.go` to properly emulate the server behavior, and complain / fail the test if the query param is not present.

File: internal/providers/gitlab/properties_test.go [hunk 2]
```diff
@@ -131,7 +131,6 @@ func Test_gitlabClient_FetchAllProperties(t *testing.T) {
 			},
 		},
 		{
-			name: "repository succeeds",
```
Comment (evankanderson) [inline internal/providers/gitlab/properties_test.go:134]: Why change this name to empty-string?

File: internal/providers/gitlab/properties_test.go [hunk 3]
```diff
@@ -143,19 +142,30 @@ func Test_gitlabClient_FetchAllProperties(t *testing.T) {
...
+					Name:                "project-1",
+					Description:         "project-1 description",
+					Visibility:          gitlab.PrivateVisibility,
+					Archived:            false,
+					ForkedFromProject:   nil,
 					Namespace: &gitlab.ProjectNamespace{
 						Path: "group",
 					},
+					License: &gitlab.ProjectLicense{
+						Name: "mit", // Changed Key to Name to match repository_properties.go
```
Comment (evankanderson) [inline internal/providers/gitlab/properties_test.go]: Will this comment make sense in 6 months?

File: internal/providers/gitlab/properties_test.go [hunk 4]
```diff
@@ -397,3 +407,44 @@ func newTestGitlabProvider(endpoint string) *gitlabClient {
 		cli: &http.Client{},
 	}
 }
+
+
+func TestGitlabProjectToProperties_License(t *testing.T) {
```
Comment (evankanderson) [inline internal/providers/gitlab/properties_test.go]: What does this test cover that the other test doesn't? 

In general, extra tests add maintenance burden when refactoring code.  That burden pays off when the tests cover additional important code cases, but redundant tests add more maintenance cost without commensurate benefits.

**Review by evankanderson — COMMENTED** [review]
Thanks for adding the tests!  The test implementation brought up a couple questions (this is the learning process, so it's totally normal to hit these sorts of things).

File: internal/providers/gitlab/properties_test.go (hunk 2 above)
Comment (Anni0506) [reply internal/providers/gitlab/properties_test.go:134]: My mistake! Restored the test name back to \"repository succeeds\".

File: internal/providers/gitlab/properties_test.go (hunk 3 above)
Comment (Anni0506) [reply internal/providers/gitlab/properties_test.go]: Good point—removed the temporary comment to keep the code clean!

File: internal/providers/gitlab/properties_test.go (hunk 4 above)
Comment (Anni0506) [reply internal/providers/gitlab/properties_test.go]: Agreed! Since the table test already covers RepoPropertyLicense, I removed this redundant test to avoid duplicate coverage and extra maintenance.

File: internal/providers/gitlab/properties_test.go [hunk 5]
```diff
@@ -143,9 +143,17 @@ func Test_gitlabClient_FetchAllProperties(t *testing.T) {
 				properties.RepoPropertyIsPrivate:  true,
 				properties.RepoPropertyIsArchived: false,
 				properties.RepoPropertyIsFork:     false,
+				properties.RepoPropertyLicense:    "mit",
 			}),
 			wantErr: false,
-			gitLabServerMockFunc: func(w http.ResponseWriter, _ *http.Request) {
+			gitLabServerMockFunc: func(w http.ResponseWriter, r *http.Request) {
+				// Verify the query parameter you added is being sent
```
Comment (evankanderson) [inline internal/providers/gitlab/properties_test.go:150]: The same comment here [as in a previous review](https://github.com/mindersec/minder/pull/6593#discussion_r3597006886) -- will this comment make sense when reading the code in 3 months?

File: internal/providers/gitlab/properties_test.go [hunk 6]
```diff
@@ -143,9 +143,17 @@ func Test_gitlabClient_FetchAllProperties(t *testing.T) {
 				properties.RepoPropertyIsPrivate:  true,
 				properties.RepoPropertyIsArchived: false,
 				properties.RepoPropertyIsFork:     false,
+				properties.RepoPropertyLicense:    "mit",
```
Comment (evankanderson) [inline internal/providers/gitlab/properties_test.go:146]: I think this is just:


```suggestion
				RepoPropertyLicense:    "mit",
```

Since it's from _this_ package, not from the shared `properties` package.

**Review by evankanderson — COMMENTED** [review]
Your test is failing to compile due to the incorrect package name for the constant.

**Comment by intelligent-ears** [comment]
We've merged #6724, which landed the same fix with the test wired into place, so I'm closing this as this seems dead.
